frontend/javascripts/oxalis/view/components/input_component.tsx

```diff
@@ -96,19 +119,31 @@ class InputComponent extends React.PureComponent<InputComponentProps, InputCompo
 
   render() {
     const { isTextArea, onPressEnter, title, style, ...inputProps } = this.props;
-    const InputComponentType: typeof TextArea | typeof Input = isTextArea ? TextArea : Input;
-    const input = (
-      <InputComponentType
+    // const InputComponentType: typeof TextArea | typeof Input = isTextArea ? TextArea : Input;
+    const input = isTextArea ? (
+      <TextArea
+        {...inputProps}
+        style={title == null ? style : undefined}
+        onChange={this.handleChange}
+        onFocus={this.handleFocus}
+        onBlur={this.handleBlur}
+        value={this.state.currentValue}
+        onPressEnter={onPressEnter}
+        onKeyDown={this.blurOnEscape}
+      />
+    ) : (
+      <Input
         {...inputProps}
         style={title == null ? style : undefined}
         onChange={this.handleChange}
         onFocus={this.handleFocus}
         onBlur={this.handleBlur}
         value={this.state.currentValue}
-        onPressEnter={onPressEnter != null ? onPressEnter : this.blurYourself}
+        onPressEnter={onPressEnter}
```

Comment:
This could probably be put into an object and then spread into here and into `TextArea` (`{ style: title == .., onChange: ... }`) to avoid the redundancy.

Reply:
Nice idea, but it seems like this is not possible. When specifying all the props individually like in this PR then TS can do its discriminative typing thing and resolve the correct type information for all the callbacks, either `EventHandler<Input>` or `EventHandler<TextArea>`. 
When shortening the props into a single object, each callback will have a mixed `EventHandler<Input|TextArea>` type and TS will complain.

Reply:
Ok, fair enough, then let's keep it as is :+1: